src/model/model_input.jl

```diff
@@ -36,35 +14,34 @@ end
     conditions.
 - `idx::Dict` of parameter indices.
 """
-
-function _model_input(year::Any, d::Dict{Symbol,DataFrame}, set::Dict, idx::Dict = Dict())
-        return _model_input(ensurearray(year), d, set, idx)
+function model_input(year::Any, d::Dict, set::Dict, idx::Dict=Dict())
+    return model_input(ensurearray(year), d, set, idx)
 end
 
-function _model_input(year::Array{Int,1}, d::Dict{Symbol,DataFrame}, set::Dict, idx::Dict = Dict())
+
+function model_input(year::Array{Int,1}, d::Dict, set::Dict, idx::Dict=Dict())
     @info("Preparing model data for $year.")
     if length(year) > 1
-        d2 = Dict(k => filter_with(df, (yr = year,); extrapolate = true) for (k,df) in d)
+        d2 = Dict(k => filter_with(df, (yr = year,); extrapolate=true) for (k, df) in d)
 
-        isempty(idx) && (idx = Dict(k => findindex(df) for (k,df) in d2))
+        isempty(idx) && (idx = Dict(k => findindex(df) for (k, df) in d2))
         (set, idx) = _model_set!(d2, set, idx)
 
-        d1 = Dict(k => filter_with(df, (yr = year,); drop = true) for (k,df) in d)
-        d1 = Dict(k => convert_type(Dict, fill_zero(set, df)) for (k,df) in d1)
+        d1 = Dict(k => filter_with(df, (yr = year,); drop=true) for (k, df) in d)
+        d1 = Dict(k => convert_type(Dict, fill_zero(df; with=set)) for (k, df) in d1)
```

Comment:
**Compatibility**. Updated fill_zero to either take a set or not. I did this so that it's more explicit zeros *in* `df` are being filled (btw @jonmbecker)